automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/BatchNodepoolCreationSpec.scala

```diff
@@ -0,0 +1,171 @@
+          )
+          getCluster = gkeServiceResource.use { gkeClient =>
+            val id = KubernetesClusterId(googleProject, LeonardoConfig.Leonardo.location, clusterName)
+            gkeClient.getCluster(id)
+          }
+
+          monitorBatchCreationResult <- testTimer.sleep(30 seconds) >> streamFUntilDone(getCluster, 20, 10 seconds)(
+            testTimer,
+            clusterDoneCheckable
+          ).compile.lastOrError
+
+          //here we sleep, because the above verifies the google state and we need to wait until leo has polled and updated its internal state to proceed
+          //it is a long time because there is a lot of stuff leo has to do besides GKE entity creation
+          _ <- testTimer.sleep(5 minutes)
+
+          _ = monitorBatchCreationResult.map(_.getNodePoolsList().size()) shouldBe Some(2)
+
+          diskConfig1 = Some(PersistentDiskRequest(DiskName("disk1"), None, None, Map.empty))
+
+          _ <- loggerIO.info("About to create app1")
+
+          _ <- LeonardoApiClient.createApp(googleProject,
+                                           appName1,
+                                           createAppRequest = defaultCreateAppRequest.copy(diskConfig = diskConfig1))
+
+          _ <- loggerIO.info("About to get app1")
+
+          getApp1 = LeonardoApiClient.getApp(googleProject, appName1)
+          monitorApp1CreationResult <- testTimer.sleep(30 seconds) >> streamFUntilDone(getApp1, 60, 10 seconds)(
+            testTimer,
+            appDoneCheckable
+          ).compile.lastOrError
+
+          _ <- loggerIO.info(s"app1 monitor result: ${monitorApp1CreationResult}")
+
+          clusterAfterApp1 <- getCluster
+          _ = clusterAfterApp1.map(_.getNodePoolsList().size()) shouldBe Some(2)
+
+          appName2 = AppName("app2")
+          diskConfig2 = Some(PersistentDiskRequest(DiskName("disk2"), None, None, Map.empty))
+
+          _ <- loggerIO.info("About to create app2")
+
+          _ <- LeonardoApiClient.createApp(googleProject,
+                                           appName2,
+                                           createAppRequest = defaultCreateAppRequest.copy(diskConfig = diskConfig2))
+
+          //creating a second app with 1 precreated nodepool should cause a second user nodepool to be created
+          getApp2 = LeonardoApiClient.getApp(googleProject, appName2)
+          monitorApp2CreationResult <- testTimer.sleep(30 seconds) >> streamFUntilDone(getApp2, 60, 10 seconds)(
+            testTimer,
+            appDoneCheckable
+          ).compile.lastOrError
+
+          _ <- loggerIO.info(s"app2 monitor result: ${monitorApp2CreationResult}")
+
+          clusterAfterApp2 <- getCluster
+          _ = clusterAfterApp2.map(_.getNodePoolsList().size()) shouldBe Some(3)
+
+          // we can only delete 1 app at a time due to the google limitation that a cluster can only have 1 nodepool related operation ongoing at a time
```

Comment:
If we do get 2 concurrent deleteApp's, does the second one poll/retry until the nodepool can be deleted? We should probably support that case since users can delete at any time

Reply:
no, it throws an error. We did say we should serialize them, but I think that work got overshadowed by the batch creation. 

We should probably do it though... I believe what would happen is it would get stuck in a 'Deleting' status from the perspective of the user